src/server/editorServices.ts

```diff
@@ -337,7 +337,8 @@ namespace ts.server {
         return `Project: ${project ? project.getProjectName() : ""} WatchType: ${watchType}`;
     }
 
-    function updateProjectIfDirty(project: Project) {
+    /*@internal*/
+    export function updateProjectIfDirty(project: Project) {
         return project.dirty && project.updateGraph();

```

Comment:
Why not put this check directly in `updateGraph`?  Are there cases where we want to force an update regardless?

Reply:
There are cases we already know that project is dirty.. 
Updategraph is also overridden to do extra things depending on project type so update graph is not place where you want to do this check

Reply:
As a caller, how do I know whether I should call `updateProjectIfDirty` or `updateGraph`?